### PR TITLE
FIT 372 SemVer For Perry

### PR DESCRIPTION
**Pull request by AVandana-OSI**
### JIRA Issue Link
- [FIT-372: Implement Semantic Versioning Support for Perry](https://osi-cwds.atlassian.net/browse/FIT-372)

### Technical Description
Built the SemVer for Perry and tested on the Jenkins with a separate Jenkins job 
http://jenkins.dev.cwds.io:8080/view/perry/job/perry-master-build-testing/

### Tests
- [ ] I have included unit tests 
- [ ] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [x] I have NOT included tests 
It's a Jenkins file

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: Jenkinsfile-complete.groovy [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
                         string(defaultValue: '', description: 'Used for mergerequest default is empty', name: 'refspec'),
-                        booleanParam(defaultValue: true, description: 'Default release version template is: <majorVersion>_<buildNumber>-RC', name: 'RELEASE_PROJECT'),
-                        string(defaultValue: "", description: 'Fill this field if need to specify custom version ', name: 'OVERRIDE_VERSION'),
                         booleanParam(defaultValue: true, description: 'Enable NewRelic APM', name: 'USE_NEWRELIC'),
+                        booleanParam(defaultValue: true, description: 'Default release version template is: <majorVersion>_<buildNumber>-RC', name: 'RELEASE_PROJECT'),
                         string(defaultValue: 'inventories/tpt2dev/hosts.yml', description: '', name: 'inventory'),
                         string(defaultValue: 'https://web.dev.cwds.io/perry', description: 'Perry base URL', name: 'PERRY_URL'),
-                ]), pipelineTriggers([pollSCM('H/5 * * * *')])])
+                ]),
+               pipelineTriggers([
```
Comment (edgibbs) [inline Jenkinsfile-complete.groovy]: had a quick discussion about using the shared library pullRequestMergedTriggerProperies.

File: Jenkinsfile-complete.groovy [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
-                        string(defaultValue: 'SNAPSHOT', description: 'Release version (if not SNAPSHOT will be released to lib-release repository)', name: 'VERSION'),
                         string(defaultValue: 'latest', description: '', name: 'APP_VERSION'),
                         string(defaultValue: 'master', description: 'perry branch', name: 'branch'),
                         string(defaultValue: 'master', description: 'ansible branch', name: 'ansible_branch'),
                         string(defaultValue: '', description: 'Used for mergerequest default is empty', name: 'refspec'),
-                        booleanParam(defaultValue: true, description: 'Default release version template is: <majorVersion>_<buildNumber>-RC', name: 'RELEASE_PROJECT'),
-                        string(defaultValue: "", description: 'Fill this field if need to specify custom version ', name: 'OVERRIDE_VERSION'),
                         booleanParam(defaultValue: true, description: 'Enable NewRelic APM', name: 'USE_NEWRELIC'),
+                        booleanParam(defaultValue: true, description: 'Default release version template is: <majorVersion>_<buildNumber>-RC', name: 'RELEASE_PROJECT'),
```
Comment (LeonidMarushevskyi) [inline Jenkinsfile-complete.groovy]: Why you left this parameter?

File: Jenkinsfile-complete.groovy [hunk 3]
```diff
@@ -96,34 +105,15 @@ node('dora-slave') {
...
             // Artifactory
             rtGradle.deployer.deployArtifacts = true
-            if (params.RELEASE_PROJECT) {
-                echo "!!!! PUSH RELEASE VERSION ${params.VERSION}"
-                buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'publish -DRelease=$RELEASE_PROJECT -DBuildNumber=$BUILD_NUMBER -DCustomVersion=$OVERRIDE_VERSION'
-            } else {
-                echo "!!!! PUSH SNAPSHOT VERSION"
-                buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'publish'
-            }
+            buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: "publish -DRelease=$RELEASE_PROJECT -DnewVersion=${newTag}".toString()
```
Comment (LeonidMarushevskyi) [inline Jenkinsfile-complete.groovy]: Just set true here, because we want to release new version every time.

Comment (AVandana-OSI) [reply Jenkinsfile-complete.groovy]: Sure

File: Jenkinsfile-complete.groovy (hunk 2 above)
Comment (AVandana-OSI) [reply Jenkinsfile-complete.groovy]: @LeonidMarushevskyi While publishing the artifacts -DRelease=$RELEASE_PROJECT is expecting or else the artifacts is not getting pushed, not sure why

**Pull request merged**